torch/export/export_schema.py

```diff
@@ -0,0 +1,331 @@
+    #   //                         Offset values SHOULD be multiples 4096 (page size) to enable mmap support.
+    #   // - "length" (optional) - number of bytes containing data. Integer stored as string.
+    #   // - "checksum" (optional) - SHA1 digest of file specified in under 'location' key.
+    #   repeated StringStringEntryProto external_data
+    location: str
+    offset: str     # !!! Consider using int, but int has int_max limitation
+    length: str     # !!! Consider using int, but int has int_max limitation
+    checksum: str
+
+
+@dataclass
+class Storage:
+    class DataLocation(Enum):
+        Default = auto()
+        External = auto()
+
+    data_location: DataLocation
+    data: Union[Buffer, ExternalBuffer]
+
+
+# This is a concrete tensor backed by storage
+@dataclass
+class Tensor:
+    # storage
+    storage: Storage
+
+    # metadata
+    meta: TensorMeta
+
+
+# !!! consider adding an optional named field to Tensor directly to avoid defining a new class
+@dataclass
+class NamedTensor:
+    name: str
+    tensor: Tensor
+
+################################################################################
+# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+
+# IValue has no corresponding class in fx
+# IValue is the "values" that are passed between nodes in the graph
+# !!! Assumption: Only Tensor can be passed between nodes, and not other int/float/bool types...
+# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
+# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+@dataclass
+class IValue:
+    name: str   # A unique identifider name for the IValue
+                # The name will be used in the graph and node to refer to the IValue
+
+    meta: TensorMeta
+
+
+# Maps to fx.Node
+@dataclass
+class Node:
+    op: str         # One of ['placeholder', 'call_function', 'get_attr', 'output'],
+                    # call_method and call_module are not supported, as they shouldn't apprear in the Caononical FX Graph
+                    # !!! Consider using an enum instead of string
+
+    target: str      # fully qualified name to the target, e.g. aten.add.Tensnor
+                     # !!! Consider using a structured operator name instead of string
+
+    args: List[Argument]              # args for this node
+    kwargs: List[KeywordArgument]     # kwargs for this node
+                                      # !!! Not all types in Argument are used as kwargs, e.g. TensorArgument should not be used as kwargs
+                                      # Do we want to enforce this in the schema? i.e. only allow certain types to be used as kwargs?
+
+    outputs: List[TensorArgument]   # A list of Tensor Argument returned by this node
+                                    # !!! Notice: this assumes that a node can only return Tensor(s), and not other int/float/bool types...
```

Comment:
this is not a correct assumption, for example `tensor.item()` returns a scalar type.

Reply:
Yes, I also realized this problem. 

Operator return types can be SymInt, but I am not sure about other scalar types. 

IIUC, tensor.item() would result in a graph break in dynamo. I don't know how dynamo will handle .item() call in export mode? 
In another word, I am not sure if we will ever have aten._local_scalar_dense (the corresponding aten op for .item)  node in a canonical FX graph. 

But I agree we will have sym_size() node in the graph. 

cc @voznesenskym